apps/condo/domains/acquiring/tasks/utils/queries.js

```diff
@@ -0,0 +1,383 @@
+    } catch (e) {
+        registerError = get(e, 'errors[0].message') || get(e, 'message')
+    }
+
+    // failed to register multi payment
+    if (!registerResponse || !registerResponse.multiPaymentId) {
+        return {
+            registered:false,
+            errorCode: PAYMENT_ERROR_CAN_NOT_REGISTER_MULTI_PAYMENT_CODE,
+            errorMessage: `Can not register multi payment: ${registerError}`,
+        }
+    }
+
+    // get multi payment
+    const multiPayment = await MultiPayment.getOne(context, {
+        id: registerResponse.multiPaymentId,
+    })
+
+    // check limits for recurrent context
+    // in case if limit is set up
+    if (recurrentContext.limit && Big(multiPayment.amount).gt(Big(recurrentContext.limit))) {
+        return {
+            registered: false,
+            errorCode: PAYMENT_ERROR_LIMIT_EXCEEDED_CODE,
+            errorMessage: `RecurrentPaymentContext limit exceeded for multi payment ${multiPayment.id}`,
+        }
+    }
+
+    return {
+        registered: true,
+        ...registerResponse,
+    }
+}
+
+async function sendResultMessageSafely (context, recurrentPayment, success, errorCode) {
+    const {
+        id,
+        tryCount,
+        recurrentPaymentContext: { id: recurrentContextId },
+    } = recurrentPayment
+    const uniqKey = `rp_${id}_${tryCount + 1}_${success}`
+    const type = success ? RECURRENT_PAYMENT_PROCEEDING_SUCCESS_RESULT_MESSAGE_TYPE
+        : RECURRENT_PAYMENT_PROCEEDING_FAILURE_RESULT_MESSAGE_TYPE
+    const meta = success ? {} : { errorCode }
+
+    try {
+        const [recurrentContext] = await getItems({
+            context,
+            listKey: 'RecurrentPaymentContext',
+            where: { id: recurrentContextId },
+            returnFields: 'serviceConsumer { resident { user { id } } }',
+        })
+        const {
+            serviceConsumer: { resident: { user: { id: userId } } },
+        } = recurrentContext
+
+        await sendMessage(context, {
+            ...dvAndSender,
+            to: { user: { id: userId } },
+            type,
+            uniqKey,
+            meta: {
+                dv: 1,
+                recurrentPaymentContext: { id: recurrentContextId },
+                ...meta,
+            },
+        })
+    } catch (error) {
+        logger.error({ msg: 'sendMessage error', error })
+    }
+}
+
+async function sendTomorrowPaymentNotificationSafely (context, recurrentPaymentContext) {
+    try {
+        const [recurrentContext] = await getItems({
+            context,
+            listKey: 'RecurrentPaymentContext',
+            where: { id: recurrentPaymentContext.id },
+            returnFields: 'serviceConsumer { resident { user { id } } }',
+        })
+        const {
+            serviceConsumer: { resident: { user: { id: userId } } },
+        } = recurrentContext
+
+        const date = dayjs()
+        const uniqKey = `rp_tp_${recurrentPaymentContext.id}_${date.toISOString()}`
```

Comment:
What if we will have an error and fix it the next day?

Reply:
I have to use trigger identifier (period || recurrentPaymentId) to make it works in such cases

Reply:
I'll update specs to test such cases